### ISSUE TITLE
FastVLM-0.5B Generate Text while didn't stop.

### ISSUE THREAD

**Issue by ptkjw1997** [apple/ml-fastvlm#64]
Thank you for open-sourcing FastVLM !
It will help me and other Multi-modal LLMs researchers a lot !

By the way, I have some problem when generating text with FastVLM-0.5B.

Below is the code of loading model.

  import torch
  from PIL import Image
  from transformers import AutoTokenizer, AutoModelForCausalLM
  MID = "apple/FastVLM-0.5B"
  IMAGE_TOKEN_INDEX = -200  # what the model code looks for
  # Load
  tok = AutoTokenizer.from_pretrained(MID, trust_remote_code=True)
  model = AutoModelForCausalLM.from_pretrained(
      MID,
      torch_dtype=torch.float16 if torch.cuda.is_available() else torch.float32,
      device_map="auto",
      trust_remote_code=True,
  )

  prompt_text = "Two points are annotated on the image, labeled by A and B. Consider the surface color of the points (the albedo of the surface, without the effect of shading). Which point has darker surface color, or the colors is about the same?\n(1) A is darker\n(2) B is darker\n(3) About the same\nAnswer with the option's number from the given choices directly."
  
  messages = [
      {"role": "user", "content": "<image>\n" + prompt_text}
  ]
  
  rendered = tok.apply_chat_template(
      messages, add_generation_prompt=True, tokenize=False
  )
  
  pre, post = rendered.split("<image>", 1)
  pre_ids  = tok(pre,  return_tensors="pt", add_special_tokens=False).input_ids
  post_ids = tok(post, return_tensors="pt", add_special_tokens=False).input_ids
  
  img_tok = torch.tensor([[IMAGE_TOKEN_INDEX]], dtype=pre_ids.dtype)
  input_ids = torch.cat([pre_ids, img_tok, post_ids], dim=1).to(model.device)
  attention_mask = torch.ones_like(input_ids, device=model.device)
  
  with torch.no_grad():
      out = model.generate(
          inputs=input_ids,
          attention_mask=attention_mask,
          images=image_tensor,
          max_new_tokens=20,
          return_dict_in_generate=True,
          output_scores=False,
          output_attentions=False,
          output_hidden_states=False,
          use_cache=True 
      )

Above code was for evaluationg BLINK Dataset.
And Generated Text is below :
    GenerateDecoderOnlyOutput(sequences=tensor([[    17, 151645,    198,  16141,     25,    425, 151645,    198,  16141,
                 25,    425, 151645,    198,  16141,     25,    425, 151645,    198,
              16141,     25]], device='cuda:0'), scores=None, logits=None, attentions=None, hidden_states=None, past_key_values=<transformers.cache_utils.DynamicCache object at 0x15411427bb90>)

'2\nAnswer: B\nAnswer: B\nAnswer: B\nAnswer:'

As I set the max_new_token = 20, it stops.
When I set the max_new_token more lareger, it repeats the above answers.
It is observed in other questions same and for model FastVLM-1.5B

I didn't check this issue occurs only for multiple-choice questions.
